pkg/platform/yaml/configdb/user/db.go

```diff
@@ -35,65 +29,27 @@ func HomeDir() string {
 	return os.Getenv("HOME") + configFolder
 }
 
-func New(logger *zap.Logger) *Db {
+func New(logger *zap.Logger, cfg *config.Config) *Db {
 	return &Db{
 		logger: logger,
+		cfg:    cfg,
 	}
 }
 
 func (db *Db) GetInstallationID(ctx context.Context) (string, error) {
 	var id string
-	id = getInstallationFromFile(db.logger)
-	if id == "" {
-		id = primitive.NewObjectID().String()
-		err := db.setInstallationID(ctx, id)
+	var err error
+	if db.cfg.InDocker {
+		id = os.Getenv("INSTALLATION_ID")
+	} else {
+		id, err = machineid.ID()
 		if err != nil {
-			return "", fmt.Errorf("failed to set installation id in file. error: %s", err.Error())
-		}
-	}
-	return id, nil
-}
-
-func getInstallationFromFile(logger *zap.Logger) string {
-	var (
-		path = HomeDir()
-		id   = ""
-	)
-
-	file, err := os.OpenFile(filepath.Join(path, "installation-id.yaml"), os.O_RDONLY, fs.ModePerm)
-	if err != nil {
-		return id
-	}
-	defer func() {
-		if err := file.Close(); err != nil {
-			utils.LogError(logger, err, "failed to close file")
+			return "", errors.New("failed to get machine id")
```

Comment:
the machine id is different in mac and lima. Hopefully this is an expected behaviour.

Reply:
yes